examples/cpp/BenchmarkFixedRadiusSearch.cpp

```diff
@@ -0,0 +1,161 @@
+// ----------------------------------------------------------------------------
+// -                        Open3D: www.open3d.org                            -
+// ----------------------------------------------------------------------------
+// The MIT License (MIT)
+//
+// Copyright (c) 2018 www.open3d.org
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+// To run, from Open3D directory:
+/* [CPU]: Open3D$ ./build/bin/examples/BenchmarkNNS CPU:0 \
+                    examples/test_data/ICP/cloud_bin_0.pcd \
+                    examples/test_data/ICP/cloud_bin_1.pcd
+  [CUDA]: Open3D$ ./build/bin/examples/BenchmarkNNS CUDA:0 \
+                    examples/test_data/ICP/cloud_bin_0.pcd \
+                    examples/test_data/ICP/cloud_bin_1.pcd
+*/
+
+#include "open3d/Open3D.h"
+using namespace open3d;
+
+// Parameters to adjust according to the test pointcloud.
+double voxel_downsample_factor = 1.0;
```

Comment:
Wei has pointed out that voxel_downsample_factor and max_correspospondence_dist were not making much sense. For large point cloud, we can take voxel_downsample_factor = 0.05, and max_correspondence_dist = 0.075.
[distance must be ~ 1.5x voxel downsample factor].

Reply:
I used  0.01 and 0.02 as cloud_bin_{0,1} is much smaller than previous data.